src/profile-logic/sanitize.js

```diff
@@ -374,11 +424,12 @@ function sanitizeThreadPII(
   }
 
   if (
-    PIIToBeRemoved.shouldRemovePrivateBrowsingData &&
-    windowIdToBeSanitized.size > 0
+    windowIdFromPrivateBrowsing.size > 0 ||
```

Comment:
Currently, this set is non-empty only when `shouldRemovePrivateBrowsingData` is true, so I simplified this test.